### PR TITLE
Implement CLI Parameter to Print Specified Object Types from OpenAPI Spec

### LINKED ISSUE #6: Print the list of objects from the spec identified by the type specified as CLI parameter
Parse an OpenAPI spec(input: URL) and print the list of objects from the spec identified by the type specified as CLI parameter.

The possible types are listed [here](https://spec.openapis.org/oas/latest.html#fixed-fields-5)
schemas
responses
parameters
examples
requestBodies
headers
securitySchemes
links
callbacks
pathItems

This is a qualifying task for GSoC'23.

### PR DESCRIPTION
**Pull request by wldeh**
Closes #6 . Addresses the qualifying task for GSoC'23 by implementing the functionality to parse an OpenAPI spec from a given URL and print the list of objects identified by the type specified as a CLI parameter. The supported object types include schemas, responses, parameters, examples, requestBodies, headers, securitySchemes, links, callbacks, and pathItems.

### CODE REVIEW

**Comment by bavulapati** [comment]
@wldeh Please take a look at the [comment](https://github.com/postman-open-technologies/openapi-componentizer/issues/6#issuecomment-1488251123) on the issue #7 

Also, you can link issues and PRs by using references in the PR description. https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

**Comment by bavulapati** [comment]
Also, take a look at https://www.conventionalcommits.org/en/v1.0.0/

**Comment by wldeh** [comment]
Thank you for your detailed feedback, @bavulapati. I reviewed the comment on issue #7 and am making the necessary changes. My condolences for not mentioning the issue in my pr. I have always strived to follow the guidelines outlined by Conventional Commits when making commits to my projects and appreciate the link you provided and will make sure to reference it in the future. Thanks again for your feedback!

File: openapi-parser/.husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
```
Comment (bavulapati) [inline openapi-parser/.husky/pre-commit:2]: Is this auto generated by husky?

File: openapi-parser/docs/README.md
Comment (bavulapati) [inline openapi-parser/docs/README.md:1]: Is this auto-generated?
Looks like main README.

File: openapi-parser/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+```bash
+git clone https://github.com/postman-open-technologies/openapi-componentizer.git
+cd openapi-parser
+npm install
+```
+
+## Usage
+
+Use `ts-node` to run the CLI tool:
```
Comment (bavulapati) [inline openapi-parser/docs/README.md:25]: Nice to have:  tsc can be put to work and run the js using node.

File: openapi-parser/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- Written in TypeScript with strong typing and error handling
+
+## Installation
+
+Clone the repository and install the dependencies:
+
+```bash
+git clone https://github.com/postman-open-technologies/openapi-componentizer.git
+cd openapi-parser
+npm install
```
Comment (bavulapati) [inline openapi-parser/README.md:18]: `npm i` fails with
```
❯ npm i

> openapi-componentizer@0.0.1 prepare
> husky install

.git can't be found (see https://git.io/Jc3F9)
npm ERR! code 1
npm ERR! path /Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser
npm ERR! command failed
npm ERR! command sh -c husky install

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/balakrishna/.npm/_logs/2023-03-30T13_36_34_943Z-debug-0.log
```

**Review by bavulapati — CHANGES_REQUESTED** [review]
Requesting changes as I couldn't test it due to npm install failure.

As you are using handful of utilities, it might be worth describing the way you use them and a way to navigate the code accordingly.

CMIIW, I notice you include tests. It's worth providing a command to run them.

**Comment by wldeh** [comment]
> Requesting changes as I couldn't test it due to npm install failure.
> 
> As you are using handful of utilities, it might be worth describing the way you use them and a way to navigate the code accordingly.
> 
> CMIIW, I notice you include tests. It's worth providing a command to run them.

Hello @bavulapati! Yes, I can definitely make the documentation as comprehensive as possible. I’ll push a commit tonight that explains each dependency and why its frequently used in a TypeScript project like this. Additionally, I’ll make sure to include instructions in the README on how to run tests and why I chose to use Jest. You can run the existing tests right now by executing the test script defined in the package.json file. I’m still working on writing robust unit tests, which takes some time. Your feedback is much appreciated.

File: openapi-parser/README.md (hunk 3 above)
Comment (wldeh) [reply openapi-parser/README.md:18]: I have addressed the issue you mentioned in my recent commit (https://github.com/postman-open-technologies/openapi-componentizer/pull/7/commits/072bd7895a6aefbee63e4111fc9ddb86e7423397). Feel free to let me know if you encounter any further issues after pulling and thank you for bringing this to my attention.

File: openapi-parser/docs/README.md (hunk 2 above)
Comment (wldeh) [reply openapi-parser/docs/README.md:25]: Hello @bavulapati. I have made a recent commit [38f0ef5](https://github.com/postman-open-technologies/openapi-componentizer/pull/7/commits/38f0ef520fba67b8ee2f7ed5e33a488a471d948f) that addresses this. Please do not hesitate to make any other suggestions.

File: openapi-parser/docs/README.md
Comment (wldeh) [reply openapi-parser/docs/README.md:1]: Yes @bavulapati , the `openapi-parser/docs/README.md` file is the original README autogenerated and modified by Typedoc to include a reference to [Exports](modules.md) at the top.

File: openapi-parser/.husky/pre-commit (hunk 1 above)
Comment (wldeh) [reply openapi-parser/.husky/pre-commit:2]: Yes, this is autogenerated by husky via the `npx husky add .husky/pre-commit "npm run pre-commit"` command

**Comment by bavulapati** [comment]
I think `npm run build` need to be run before running the cli, at the moment.
Ran into the following 

```
ESM Build start
✘ [ERROR] Could not resolve "node-fetch"

    src/openapi/fetcher.ts:1:18:
      1 │ import fetch from 'node-fetch'
        ╵                   ~~~~~~~~~~~~

  You can mark the path "node-fetch" as external to exclude it from the bundle, which will remove
  this error.

ESM Build failed
Error: Build failed with 1 error:
src/openapi/fetcher.ts:1:18: ERROR: Could not resolve "node-fetch"
    at failureErrorWithLog (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1636:15)
    at /Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1048:25
    at runOnEndCallbacks (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1471:45)
    at buildResponseToResult (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1046:7)
    at /Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1075:16
    at responseCallbacks.<computed> (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:697:9)
    at handleIncomingPacket (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:752:9)
    at Socket.readFromStdout (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:673:7)
    at Socket.emit (node:events:512:28)
    at addChunk (node:internal/streams/readable:324:12)
```

**Comment by wldeh** [comment]
> I think `npm run build` need to be run before running the cli, at the moment. Ran into the following
> 
> ```
> ESM Build start
> ✘ [ERROR] Could not resolve "node-fetch"
> 
>     src/openapi/fetcher.ts:1:18:
>       1 │ import fetch from 'node-fetch'
>         ╵                   ~~~~~~~~~~~~
> 
>   You can mark the path "node-fetch" as external to exclude it from the bundle, which will remove
>   this error.
> 
> ESM Build failed
> Error: Build failed with 1 error:
> src/openapi/fetcher.ts:1:18: ERROR: Could not resolve "node-fetch"
>     at failureErrorWithLog (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1636:15)
>     at /Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1048:25
>     at runOnEndCallbacks (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1471:45)
>     at buildResponseToResult (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1046:7)
>     at /Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:1075:16
>     at responseCallbacks.<computed> (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:697:9)
>     at handleIncomingPacket (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:752:9)
>     at Socket.readFromStdout (/Users/balakrishna/projects/postman-open-technologies/openapi-componentizer/wldeh/openapi-componentizer/openapi-parser/node_modules/esbuild/lib/main.js:673:7)
>     at Socket.emit (node:events:512:28)
>     at addChunk (node:internal/streams/readable:324:12)
> ```

Hello @bavulapati. I was swamped with work today and couldn’t provide an update. I’ll follow up with you by the 31st and look to re-request. Thank you for your timely feedback; I appreciate it greatly.

**Comment by wldeh** [comment]
Hello @bavulapati! I wanted to inform you that I have just addressed most of the changes that you mentioned. Please let me know if you have any questions or concerns. Thank you!

**Pull request closed without merging**